apps/web/src/components/quick-start/digest-demo-flow/DigestNode.tsx

```diff
@@ -51,17 +52,19 @@ export function DigestNode({ data, id }: { data: any; id: string }) {
 
   const [seconds, setSeconds] = useState(0);
   const interval = useInterval(() => setSeconds((sec) => sec - 1), 1000);
+  const intervalRef = useDataRef({ interval, digestInterval });
 
   useEffect(() => {
+    const { interval: intervalObject, digestInterval: currentDigestInterval } = intervalRef.current;
     if (isRunningDigest) {
-      setSeconds(digestInterval);
-      interval.start();
+      setSeconds(currentDigestInterval);
+      intervalObject.start();
     } else {
-      interval.stop();
+      intervalObject.stop();
     }
 
-    return interval.stop;
-  }, [isRunningDigest]);
+    return intervalObject.stop;
+  }, [isRunningDigest, intervalRef]);
```

Comment:
to make this change safe I do use ref to hold the `interval, digestInterval` values